src/traces/indicator/plot.js

```diff
@@ -0,0 +1,788 @@
+        title
+            .attr('text-anchor', function() {
+                return isBullet ? anchor.right : anchor[trace.title.align];
+            })
+            .text(trace.title.text)
+            .call(Drawing.font, trace.title.font)
+            .call(svgTextUtils.convertToTspans, gd);
+
+        // Position title
+        title.attr('transform', function() {
+            var titleX = size.l + size.w * position[trace.title.align];
+            var titleY;
+            var titlePadding = cn.titlePadding;
+            var titlebBox = Drawing.bBox(title.node());
+            if(hasGauge) {
+                if(isAngular) {
+                    // position above axis ticks/labels
+                    var bBox = Drawing.bBox(angularaxisLayer.node());
+                    titleY = (bBox.top - titlePadding) - titlebBox.bottom;
+                }
+                if(isBullet) {
+                    // position outside domain
+                    titleY = numbersY - (titlebBox.top + titlebBox.bottom) / 2;
+                    titleX = size.l - cn.bulletPadding * size.w; // Outside domain, on the left
+                }
+            } else {
+                // position above numbers
+                titleY = (trace._numbersTop - titlePadding) - titlebBox.bottom;
+            }
+            return strTranslate(titleX, titleY);
+        });
+    });
+};
+
+function drawBulletGauge(gd, plotGroup, cd, gaugeOpts) {
+    var trace = cd[0].trace;
+
+    var bullet = gaugeOpts.bulletGauge;
+    var bulletaxis = gaugeOpts.bulletaxisLayer;
+    var gaugeBg = gaugeOpts.gaugeBg;
+    var gaugeOutline = gaugeOpts.gaugeOutline;
+    var size = gaugeOpts.size;
+    var domain = trace.domain;
+
+    var hasTransition = gaugeOpts.hasTransition;
+    var transitionOpts = gaugeOpts.transitionOpts;
+    var onComplete = gaugeOpts.onComplete;
+
+    // preparing axis
+    var ax, vals, transFn, tickSign, shift;
+    var opts = trace.gauge.axis;
+
+    // Enter bullet, axis
+    bullet.enter().append('g').classed('bullet', true);
+    bullet.attr('transform', 'translate(' + size.l + ', ' + size.t + ')');
+
+    bulletaxis.enter().append('g')
+        .classed('bulletaxis', true)
+        .classed('crisp', true);
+    bulletaxis.selectAll('g.' + 'xbulletaxis' + 'tick,path').remove();
+
+    // Draw bullet
+    var bulletHeight = size.h; // use all vertical domain
+    var innerBulletHeight = trace.gauge.value.thickness * bulletHeight;
+    var bulletLeft = domain.x[0];
+    var bulletRight = domain.x[0] + (domain.x[1] - domain.x[0]) * ((trace._hasNumber || trace._hasDelta) ? (1 - cn.bulletNumberDomainSize) : 1);
+
+    ax = mockAxis(gd, opts, [trace.vmin, trace.vmax]);
+    ax._id = 'xbulletaxis';
+    ax.domain = [bulletLeft, bulletRight];
+    ax.setScale();
+
+    vals = Axes.calcTicks(ax);
+    transFn = Axes.makeTransFn(ax);
+    tickSign = Axes.getTickSigns(ax)[2];
+
+    shift = size.t + size.h;
+    if(ax.visible) {
```

Comment:
I don't think `visible` is part of the `gauge.axis` attribute currently. Might be a nice addition.

Reply:
Added in e6d8525ab07c56ca10bdbac6762b06d1d57bf830

Reply:
We should add a mock here showing off `gauge.axis.visible: false` for both gauge shapes.